example-cloud-connector/starter/src/main/java/org/activiti/cloud/examples/connectors/ExampleConnector.java

```diff
@@ -49,34 +49,73 @@ public class ExampleConnector {
     //just a convenience - not recommended in real implementations
     private String var1Copy = "";
 
-    @Autowired
-    private ConnectorProperties connectorProperties;
+    private final ConnectorProperties connectorProperties;
 
     private final ObjectMapper objectMapper;
 
     private final IntegrationResultSender integrationResultSender;
 
-    public ExampleConnector(IntegrationResultSender integrationResultSender, ObjectMapper objectMapper) {
+    @Autowired
+    public ExampleConnector(
+        ConnectorProperties connectorProperties,
+        IntegrationResultSender integrationResultSender,
+        ObjectMapper objectMapper
+    ) {
+        this.connectorProperties = connectorProperties;
         this.objectMapper = objectMapper;
         this.integrationResultSender = integrationResultSender;
     }
 
-    @StreamListener(value = ExampleConnectorChannels.EXAMPLE_CONNECTOR_CONSUMER)
-    public void performTask(IntegrationRequest event) throws InterruptedException {
+    @Override
+    public Void apply(IntegrationRequest event) {
+        performTask(event);
+        return null;
+    }
+
+    public void performTask(IntegrationRequest event) {
```

Comment:
The apply method can be implemented directly or via separate Connector function bean that uses ExampleConnector consumer as a delegate.